### DISCUSSION TITLE
Question about the data

### DISCUSSION THREAD

**Discussion by jtannenba** [olikraus/u8g2#2355] [Q&A] [answered]
If I were sending a short string such as "JOE" to the I2C SSD1306, in pulseview with a Salae clone, I see 3 columns of data sent, but not my 4A,4F,45.  How does the library juggle the data to send to the display over i2c?
My program works as desired, but I can't see the data over the i2c as I expect.

Thank you,
Joe
[Program.txt](https://github.com/olikraus/u8g2/files/13966204/Program.txt)

**Comment by olikraus** [comment, marked as answer]
>  How does the library juggle the data to send to the display over i2c?

U8g2 will convert your ascii chars into a bitmap and send the bitmap data. This is, why you have to assign a font.  
The font contains a bitmap image for each char.

> but I can't see the data over the i2c as I expect.

As expected??? The SSD1306 expects bitmaps not ascii chars. Just read the SSD1306 datasheet then you will understand that the expected data of the SSD1306 is bitmap data. Don't get confused with those 16x2 dot matrix displays. They do expect ascii and do the conversion to dot matrix bitmaps internally in the hardware. However this is not at all true for SSD1306.

And this is also the reason why libs like u8g2 are MUCH more complicated than LiquidChrystal lib.

**Comment by jtannenba** [comment]
I thought that when I saw the font files.

Thank you,
Joe
